tests/test_callable_view.py

```diff
@@ -94,3 +94,33 @@ def __call__(self, c: bool) -> bool:
 
     fn_view1 = CallableView.from_callable(foo, include_extras=True)
     assert fn_view1.parameters == (ParameterView("c", TypeView(bool)),)
+
+
+def test_instance_method() -> None:
+    class Foo:
+        def method(self, c: bool) -> bool:
+            return c
+
+    foo = Foo()
+
+    fn_view1 = CallableView.from_callable(foo.method, include_extras=True)
+    assert fn_view1.parameters == (ParameterView("c", TypeView(bool)),)
+
+
+@pytest.mark.parametrize(
+    ("hint",),
+    [
+        (Optional[str],),
+        (Union[str, None],),
+        (Union[str, int, None],),
+        (Optional[Union[str, int]],),
+        (Union[str, int],),
+        (str,),
+    ],
+)
+def test_parameters_with_none_default(hint: Any) -> None:
+    def fn(plain: hint = None, annotated: Annotated[hint, ...] = None) -> None: ...
+
+    fn_view = CallableView.from_callable(fn, include_extras=True)
+    plain_param, annotated_param = fn_view.parameters
+    assert plain_param.type_view.annotation == annotated_param.type_view.annotation
```

Comment:
This demos something we've built a workaround for in Litestar for versions of python < 3.11.

See https://github.com/litestar-org/litestar/pull/2516

The last two test cases fail:

```
FAILED tests/test_callable_view.py::test_parameters_with_none_default[hint4] - assert typing.Union[...int, NoneType] == typing.Union[str, int]
FAILED tests/test_callable_view.py::test_parameters_with_none_default[str] - AssertionError: assert typing.Optional[str] == <class 'str'>
```

The workaround is a bit of a monstrosity: https://github.com/litestar-org/litestar/blob/f80b4d352a79837423547e5834200396f45b7c2d/litestar/utils/signature.py#L50